linera-core/src/unit_tests/client_tests.rs

```diff
@@ -1643,76 +1680,1188 @@ where
+        .await;
+    assert!(b1_result.is_err());
+
+    let manager = client3
+        .chain_info_with_manager_values()
+        .await
+        .unwrap()
+        .manager;
+    // Validator 3 may or may not have processed the validated block before the update was
+    // canceled due to the errors from the faulty validators. Submit it again to make sure
+    // it's there, so that client 2 can download and re-propose it later.
+    let validated_block_certificate = *manager.requested_locked.unwrap();
+    let resubmission_result = builder
+        .node(3)
+        .handle_certificate(
+            validated_block_certificate,
+            Vec::new(),
+            Vec::new(),
+            CrossChainMessageDelivery::Blocking,
+        )
+        .await;
+    assert!(resubmission_result.is_err());
+
+    let validator_manager = builder
+        .node(3)
+        .chain_info_with_manager_values(chain_id3)
+        .await
+        .unwrap()
+        .manager;
+    assert!(validator_manager
+        .requested_proposed
+        .clone()
+        .unwrap()
+        .content
+        .block
+        .operations
+        .contains(&Operation::System(SystemOperation::ReadBlob {
+            blob_id: blob2_id
+        })));
+    assert!(validator_manager
+        .requested_proposed
+        .unwrap()
+        .content
+        .block
+        .operations
+        .contains(&Operation::System(SystemOperation::PublishBlob {
+            blob_id: blob3_id
+        })));
+    assert!(validator_manager
+        .requested_locked
+        .clone()
+        .unwrap()
+        .value()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::ReadBlob {
+            blob_id: blob2_id
+        })));
+    assert!(validator_manager
+        .requested_locked
+        .unwrap()
+        .value()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::PublishBlob {
+            blob_id: blob3_id
+        })));
+
+    builder.set_fault_type([1], FaultType::Offline).await;
+    builder.set_fault_type([0, 2, 3], FaultType::Honest).await;
+
+    client4.synchronize_from_validators().await.unwrap();
+    let bt_certificate = client4
+        .burn(None, Amount::from_tokens(1), UserData::default())
+        .await
+        .unwrap()
+        .unwrap();
+
+    let hashed_certificate_values = client4
+        .read_hashed_certificate_values_downward(bt_certificate.value.hash(), 4)
+        .await
+        .unwrap();
+
+    // Latest block should be the burn
+    assert!(hashed_certificate_values[0]
+        .inner()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::Transfer {
+            owner: None,
+            recipient: Recipient::Burn,
+            amount: Amount::from_tokens(1),
+            user_data: UserData::default(),
+        })));
+
+    // Block before that should be b1
```

Comment:
Did we define "b1" anywhere? Otherwise maybe describe when it was proposed.

Reply:
Yes, we have `b1_result` to indicate that that block is `b1`